README.md

```diff
@@ -51,24 +51,28 @@ The examples below are provided in TypeScript syntax. The package does work unde
 
 ### Message Bundle Loading
 
-`@dojo/i18n` provides a means for loading locale-specific messages, and updating those messages when the locale changes. Each bundle has a default module that is `import`ed like any other TypeScript module. Locale-specific messages are then loaded via the `i18n` method. Every default bundle MUST provide a `bundlePath` that will be used to determine locale bundle locations, a `locales` array of supported locales, and a `messages` map of default messages. For example, suppose the module located at `nls/common.ts` contains the following contents:
+`@dojo/i18n` provides a means for loading locale-specific messages, and updating those messages when the locale changes. Each bundle has a default module that is `import`ed like any other TypeScript module. Locale-specific messages are then loaded via the `i18n` method. Every default bundle MUST provide a `messages` map containing default messages, and an optional `locales` object that maps supported locales to functions that load their respective translations. Further, each default bundle is assigned a unique `id` property that is used internally to manage caching and handle interoperability with Globalize.js (see below). While it is possible to include an `id` with your message bundles, doing so is neither necessary nor recommended.
 
 ```typescript
-const bundlePath = 'nls/common';
-const locales = [ 'fr', 'ar', 'ar-JO' ];
-const messages = {
-	hello: 'Hello',
-	goodbye: 'Goodbye'
-};
+import fr from './fr/common';
 
-export default { bundlePath, locales, messages };
-```
+export default {
+	locales: {
+		// Locale providers can load translations lazily...
+		ar: () => import('./ar/main'),
+		'ar-JO': () => import('./ar-JO/main'),
```

Comment:
Should these be coming from /common instead of /main, similar to `./fr/common` above?